### ISSUE TITLE
Add API for firing a click event on the Upload button

### ISSUE THREAD

**Issue by vaadin-bot** [vaadin/framework#1638]
**Originally by @johndevs**

___
Currently you cannot open up the file upload dialog for an immediate upload unless you click on the upload button. 

In some cases you might want to artificially trigger that click event, for instance if you want to start the upload by selecting a menu item. Currently this is not possible.

___
 Imported from https://dev.vaadin.com/ issue #7197

**Comment by chvndb** [comment]
Any more work done on this issue?

**Comment by estefanoshin** [comment]
I just implemented a Js injection on the element:

```
String script = "document.getElementByClassName('nameOfTheCustomizedElementClass')[" +
fileUploadNumber + "].setAttribute(`document.getElementsByClassName('gwt-FileUpload')[" +
fileUploadNumber + "].click()`))";

JavaScript.eval(script);
```

Basically, you can't click the input directly so instead, you could add the onClick attribute to the customized button, div or whatever to click it for you.

The thing is that, you might need the fileUploadNumber  integer and check with the browser console if you are correctly selecting the right component.

Also, you might wanna check if the ( ` ) quote works in old browsers

**Comment by TatuLund** [comment]
Closign as won't fix as this deemed impossible with certain browsers.
